fix(core): Update `async-graphql`, `axum` and `tonic` (#6901)

kodemartin opened the pull request:
# Description of change

This patch introduces the necessary changes to restore `graphiql` endpoints.

- Updated `async-graphql` to `7.0.17`
- Updated `axum` to `0.8`
- Updated `tonic` to `0.13`
- Updated `tonic-rustls` to `0.2`
- Refactored accordingly the codebase

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

### Release Notes

- [x] GraphQL: Fix graphiql web interface

marc2332 committed:
fix(core): Update `async-graphql`, `axum` and `tonic` (#6901)

Closes https://github.com/iotaledger/iota/issues/6886

- Updated `async-graphql` to the latest commit with the CI ✅ 
- Updated `axum` to 0.8
- Updated `tonic` to 0.13
- Updated `tonic-rustls` to 0.2
- Removed the `{` -> `:` path patcher as `{}` is the correct syntax in
`axum` `>0.8`
- Moved from `/:` to `/{x}` syntax in axum endpoints
- Updated some tests
- Replaced `tonic::body::BoxBody` with `tonic::body::Body` 
- Removed the usage of `async_trait` in axum in impls of `FromRequest`
and `FromRequestParts` due to the recent stabilization of `async fn` in
traits (my guess tbh)
- Changed from `Option` to `Result` a usage of `ConnectInfo` as that
seems to be way it works now
(https://github.com/tokio-rs/axum/discussions/3245#discussioncomment-12389573)


This was necessary as `async-graphql` recently merged a fix for #6886,
but to use it we also need to update `tonic` and therefore `axum` and
`tonic-rustls`. Also, better to have more up to date deps if that means
easier to fix other issues in the future


![image](https://github.com/user-attachments/assets/4050f2f0-1d8f-4b14-86b1-2a660aeb2061)

Thoralf-M reviewed (commented):
Should be updated after https://github.com/iotaledger/iota/pull/6989

sergiupopescu199 reviewed (approved):
lgtm ✨

marc2332 committed:
refactor(code): Update `async-graphql` to `7.0.17` (#6989)

Closes https://github.com/iotaledger/iota/issues/6914

Uses the latest release of `async-graphql` which now also contains the
graphiql fix

More info: https://github.com/iotaledger/iota/pull/6901

marc2332 committed:
fix(node): Update graphql test snapshot

marc2332 committed:
fix(ts-sdk): Sort expected transactions in graphql test to avoid any order difference

marc2332 reviewed (approved):
✅ 🫡
